packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -523,6 +525,13 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
   /// See [FocusManager] for more details.
   FocusManager get focusManager => _buildOwner!.focusManager;
 
+  /// A delegate that communicates with a platform plugin for serializing and
+  /// managing platform-rendered menu bars created by [PlatformMenuBar].
+  ///
+  /// This is set by default to a [DefaultPlatformMenuDelegate] instance in
+  /// [initInstances].
+  late PlatformMenuDelegate platformMenuDelegate;
```

Comment:
So basically, PlatformMenuBar would have to listen to focus changes and when the window containing a PlatformMenuBar loses focus, it would unlock the delegate so that the PlatformMenuBar in the window gaining the focus can lock it? As long as that's possible without strange flickering of the menu bar that seems fine.